evaluator.lua

```diff
@@ -834,7 +844,7 @@ end
 
 ---@param inferrable_term inferrable
 ---@param typechecking_context TypecheckingContext
----@return value, unknown, typed
+---@return value, Array, typed
 function infer(
```

Comment:
Further down in this code (not visible in diff), we have several instance of a `usages_array` variable that does not exist anywhere, and is thus nil. This happens with several variables: `handler_type`, `backing_type` and also a function `runtime_context` is passed into something that wants an actual table, not a function.

In addition, this line throws type errors because the first and third arguments appear to be swapped:

		local stype, susages, sterm = check(terms.value.prim_bool_type, typechecking_context, subject)